confgenerator/feature_tracking.go

```diff
@@ -141,6 +146,7 @@ func trackedMappedComponents[C Component](module string, kind string, m map[stri
 	return features, nil
 }
 
+// TODO: add time.Duration to auto tracking
 func trackingFeatures(c reflect.Value, m metadata, feature Feature) ([]Feature, error) {
```

Comment:
Would you be open to renaming `m` to `md`? `m` on its own feels weirder now that tracking `map` is supported, feels like it needs more distinction.